liquidity/autoloop_test.go

```diff
@@ -1235,6 +1250,175 @@ func TestAutoLoopRecurringBudget(t *testing.T) {
 	c.stop()
 }
 
+// TestEasyAutoloop tests that the easy autoloop logic works as expected. This
+// involves testing that channels are correctly selected and that the balance
+// target is successfully met.
+func TestEasyAutoloop(t *testing.T) {
+	defer test.Guard(t)
+
+	// We need to change the default channels we use for tests so that they
+	// have different local balances in order to know which one is going to
+	// be selected by easy autoloop.
+	easyChannel1 := lndclient.ChannelInfo{
+		Active:        true,
+		ChannelID:     chanID1.ToUint64(),
+		PubKeyBytes:   peer1,
+		LocalBalance:  95000,
+		RemoteBalance: 0,
+		Capacity:      100000,
+	}
+
+	easyChannel2 := lndclient.ChannelInfo{
+		Active:        true,
+		ChannelID:     chanID1.ToUint64(),
+		PubKeyBytes:   peer1,
+		LocalBalance:  75000,
+		RemoteBalance: 0,
+		Capacity:      100000,
+	}
+
+	var (
+		channels = []lndclient.ChannelInfo{
+			easyChannel1, easyChannel2,
+		}
+
+		params = Parameters{
+			Autoloop:                  true,
+			AutoFeeBudget:             36000,
+			AutoFeeRefreshPeriod:      time.Hour * 3,
+			AutoloopBudgetLastRefresh: testBudgetStart,
+			MaxAutoInFlight:           2,
+			FailureBackOff:            time.Hour,
+			SweepConfTarget:           10,
+			HtlcConfTarget:            defaultHtlcConfTarget,
+			EasyAutoloop:              true,
+			EasyAutoloopTarget:        75000,
+			FeeLimit:                  defaultFeePortion(),
+		}
+	)
+
+	c := newAutoloopTestCtx(t, params, channels, testRestrictions)
+	c.start()
+
+	var (
+		maxAmt = 50000
+
+		chan1Swap = &loop.OutRequest{
+			Amount:          btcutil.Amount(maxAmt),
+			OutgoingChanSet: loopdb.ChannelSet{easyChannel1.ChannelID},
+			Label:           labels.AutoloopLabel(swap.TypeOut),
+			Initiator:       autoloopSwapInitiator,
+		}
+
+		quotesOut1 = []quoteRequestResp{
+			{
+				request: &loop.LoopOutQuoteRequest{
+					Amount: btcutil.Amount(maxAmt),
+				},
+				quote: &loop.LoopOutQuote{
+					SwapFee:      1,
+					PrepayAmount: 1,
+					MinerFee:     1,
+				},
+			},
+		}
+
+		loopOut1 = []loopOutRequestResp{
+			{
+				request: chan1Swap,
+				response: &loop.LoopOutSwapInfo{
+					SwapHash: lntypes.Hash{1},
+				},
+			},
+		}
+	)
+
+	// We expected one max size swap to be dispatched on our channel with
+	// the biggest local balance.
+	step := &easyAutoloopStep{
+		minAmt:      1,
+		maxAmt:      50000,
+		quotesOut:   quotesOut1,
+		expectedOut: loopOut1,
+	}
+
+	c.easyautoloop(step, false)
+
+	// In order to reflect the change on the channel balances we create a
+	// new context and restart the autolooper.
+	easyChannel1.LocalBalance -= chan1Swap.Amount
+	channels = []lndclient.ChannelInfo{
+		easyChannel1, easyChannel2,
+	}
+
+	c = newAutoloopTestCtx(t, params, channels, testRestrictions)
```

Comment:
Shouldn't we first stop the existing context? (so we also check that no unexpected error happened)